### PR TITLE
feat(ONYX-369): navigate to create alerts instead of filers

### PR DESCRIPTION
**Pull request by MounirDhahri**
This PR resolves [ONYX-369] 

### Description

- Navigate to create alerts instead of filers on "Create alert" tap when ff is enabled.

https://github.com/artsy/eigen/assets/11945712/4e4e11c9-a8c6-4a63-9eed-172a820902ec

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [X] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- navigate to create alert modal on create alert tap - mounir

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[ONYX-369]: https://artsyproduct.atlassian.net/browse/ONYX-369?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (navigate to create alert modal on create alert tap - mounir)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 2a383baad7df8d5ae7dd70711738a6fd678ebbd7
</p>

File: src/app/Components/Artist/ArtistArtworks/ArtistArtworks.tsx [hunk 1]
```diff
@@ -107,6 +119,10 @@ const ArtworksGrid: React.FC<ArtworksGridProps> = ({
     })
   }
 
+  const handleCompleteSavedSearch = () => {
+    // TODO: Get the new count of the artist saved alerts
+  }
```
Comment (MounirDhahri) [inline src/app/Components/Artist/ArtistArtworks/ArtistArtworks.tsx:124]: https://artsyproduct.atlassian.net/browse/ONYX-412

File: src/app/Scenes/SavedSearchAlertsList/SavedSearchAlertsList.tsx [hunk 2]
```diff
@@ -22,7 +22,7 @@ export const SavedSearchAlertsListQueryRenderer: React.FC<Props> = ({ artistID }
         }
       `}
       variables={{
-        artistIDs: [artistID],
+        artistIDs: artistID ? [artistID] : [],
```
Comment (MounirDhahri) [inline src/app/Scenes/SavedSearchAlertsList/SavedSearchAlertsList.tsx:25]: this was causing a local error in __DEV__: fixed it here. maybe I should have done it separately

**Comment by MounirDhahri** [comment]
I need to merge this to make sure Daria can pick the remaining tasks - glad to address feedback in a separate PR

**Pull request merged**